File: Week_01/G20200343030632/Leetcode_283_632.py
#
# @lc app=leetcode.cn id=283 lang=python3
#
# [283] 移动零
#

# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        zero = 0 # 标记值为0的下标
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[i], nums[zero] = nums[zero], nums[i]
                zero += 1

# @lc code=end

File: Week_01/G20200343030632/Leetcode_283_632.java

class Solution {
    public void moveZeroes(int[] nums) {
        int zero = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                nums[zero++] = nums[i];
            }
        }
        while (zero < nums.length) {
            nums[zero++] = 0;
        }        
    }
}